sql/core/src/test/resources/sql-tests/results/transform.sql.out

```diff
@@ -638,3 +639,15 @@ struct<b:string>
 1
 2
 3
+
+
+-- !query
+SELECT TRANSFORM(distinct b, a, c)
```

Comment:
ah the parser treats it as `distinct as b`.

Can we improve the parser to recognize `distinct` correctly and give a clear error message?

Reply:
BTW why do we support `TRANSFORM(a as b)`? is it from Hive? Normal function parameters can't have alias.

Reply:
> BTW why do we support `TRANSFORM(a as b)`? is it from Hive? Normal function parameters can't have alias.

Hive not support this. 

![image](https://user-images.githubusercontent.com/46485123/114645556-4a7ff400-9d0c-11eb-91eb-bc679ea0039a.png)

may be we should not use 
```
namedExpressionSeq
    : namedExpression (',' namedExpression)*
    ;

(SELECT kind=TRANSFORM '(' namedExpressionSeq ')'
            | kind=MAP namedExpressionSeq
            | kind=REDUCE namedExpressionSeq)
```

maybe 
```
expressionSeq 
    : expression(',' expression) 
    ;

(SELECT kind=TRANSFORM '(' expressionSeq ')'
            | kind=MAP expressionSeq
            | kind=REDUCE eExpressionSeq)
```

Reply:
yea, let's fix it.

Reply:
> yea, let's fix it as well.

Should I fix both issue in this pr or  in separated PR?

Reply:
let's have 2 PRs. forbid distinct and forbid alias are two things.

Reply:
> let's have 2 PRs. forbid distinct and forbid alias are two things.

I have make a new ticket for this one. and changed the title.

How about current